src/compiler.rs

```diff
@@ -324,8 +324,14 @@ impl AmberCompiler {
 
     #[cfg(not(windows))]
     fn find_bash() -> Option<Command> {
-        let mut command = Command::new("/usr/bin/env");
-        command.arg("bash");
-        Some(command)
+        if env::var("GITHUB_ACTIONS_BASH_CONTAINER").is_ok() {
+            let mut command = Command::new("docker");
+            command.args(["exec", "--workdir", "/root", "--user", "405", "test_container", "bash"]);
+            Some(command)
+        } else {
+            let mut command = Command::new("/usr/bin/env");
+            command.arg("bash");
+            Some(command)
+        }
```

Comment:
@lens0021 about this?

Reply:
> This should be better documented

I've added a function documentation. "Return bash command. In some situations, mainly for testing purposes, this can return a command, for example, containerized execution which is not bash but behaves like bash." 441fea972c9be450ccaad19c81605bb20d0b946a

> (Change suggestion)

Committed as 4b969598ee446d41303b6543c7639e36ff59aade. Thanks

> Do we skip this possibility that we may want to run the tests in CI workflow without any docker container?

I do not understand this sentence. Should I remove the docker approach from this PR?

> I want to test the compiler locally with the docker container.

I will write an amber script for this... I am trying
![image](https://github.com/user-attachments/assets/04907698-a709-4579-a38c-a15da3754a5f)

Reply:
@Ph0enixKM